src/web/builds.rs

```diff
@@ -111,6 +122,81 @@ pub(crate) async fn build_list_json_handler(
         .into_response())
 }
 
+async fn build_trigger_check(
+    mut conn: DbConnection,
+    name: &String,
+    version: &Version,
+    build_queue: &Arc<BuildQueue>,
+) -> AxumResult<impl IntoResponse> {
+    let _ = CrateDetails::new(&mut *conn, &name, &version, None, vec![])
```

Comment:
after having slept over it I believe we should just do a plain SQL statement here instead of fetching the full crate details. 

```
    let row = sqlx::query!(
        r#"
        SELECT id 
        FROM releases
        INNER JOIN crates ON crates.id = releases.crate_id
        WHERE crates.name = $1 AND releases.version = $2"#,
        name,
        version.to_string(),
    )
    .fetch_optional(&mut *conn)
    .await?
```

Reply:
I've used `count(*)` instead of `id` as the latter was ambiguous. Although I realize that count(*) could be a bit slower (probably depends on whether it batches the returned ids or not, and hence whether it can short-cut query evaluation). Hopefully it doesn't matter in practice.